instance/models/instance.py

```diff
@@ -17,656 +17,112 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 """
-Instance app models - Instance
+Instance app models - Open EdX Instance and AppServer models
 """
-
-# Imports #####################################################################
-
 import logging
-import string
 
-from django.conf import settings
+from django.contrib.contenttypes.fields import GenericForeignKey, GenericRelation
+from django.contrib.contenttypes.models import ContentType
 from django.db import models
-from django.db.backends.utils import truncate_name
-from django.template import loader
-from django.utils import timezone
-from django.utils.crypto import get_random_string
+from django.utils.functional import cached_property
 from django_extensions.db.models import TimeStampedModel
-from functools import partial
 
-from instance.gandi import GandiAPI
 from instance.logger_adapter import InstanceLoggerAdapter
-from instance.logging import log_exception
-from instance.models.mixins.ansible import AnsibleInstanceMixin
-from instance.models.mixins.database import MongoDBInstanceMixin, MySQLInstanceMixin, SwiftContainerInstanceMixin
-from instance.models.mixins.utilities import EmailInstanceMixin
-from instance.models.mixins.version_control import GitHubInstanceMixin
-from instance.models.utils import (
-    ModelResourceStateDescriptor, ResourceState, SteadyStateException, ValidateModelMixin
-)
-
-# Constants ###################################################################
-
-PROTOCOL_CHOICES = (
-    ('http', 'HTTP - Unencrypted clear text'),
-    ('https', 'HTTPS - Encrypted'),
-)
-
-gandi = GandiAPI()
+from .utils import ValidateModelMixin
```

Comment:
@itsjeyd True. I used some because I like relative imports, and I found the "from instance" imports to be extremely confusing when there is a file called 'instance.py' in the same folder as well as a top-level app called 'instance'.

However, I did suspect that I may be going against the practices for this codebase, given the lack of other relative imports. @antoviaque do you have a rule against them?